CodePushPackage.m

```diff
@@ -204,15 +204,22 @@ + (void)downloadPackage:(NSDictionary *)updatePackage
            failCallback:(void (^)(NSError *err))failCallback
 {
     NSString *newPackageFolderPath = [self getPackageFolderPath:updatePackage[@"packageHash"]];
-    NSError *error = nil;
+    NSError *error;
     
-    if (![[NSFileManager defaultManager] fileExistsAtPath:newPackageFolderPath]) {
-        [[NSFileManager defaultManager] createDirectoryAtPath:newPackageFolderPath
+    if (![[NSFileManager defaultManager] fileExistsAtPath:[self getCodePushPath]]) {
```

Comment:
Well, yes. Though the expectation is that the `else if` only ever saves 1 file existence check during the first time this is called. Once the CodePush path is created, it is never removed unless a new binary is deployed.